### PR TITLE
ref(replays): Rename the type used for list page results

### PR DESCRIPTION
**Pull request by ryan953**
Quick rename

This is going to create conflicts with https://github.com/getsentry/sentry/pull/37134 so i'm happy to wait.

### CODE REVIEW

File: static/app/views/replays/types.tsx [hunk 1]
```diff
@@ -2,7 +2,7 @@ import type {eventWithTime} from 'rrweb/typings/types';
 
 import type {RawCrumb} from 'sentry/types/breadcrumbs';
 
-export type Replay = {
+export type ReplayDiscoveryListItem = {
```
Comment (billyvg) [inline static/app/views/replays/types.tsx:5]: Probably don't need `Discover` here as it's an implementation detail (also we are going to have replay-specific endpoints)

```suggestion
export type ReplayListItem = {
```

**Review by billyvg — APPROVED** [review]

File: static/app/views/replays/types.tsx (hunk 1 above)
Comment (ryan953) [reply static/app/views/replays/types.tsx:5]: i named it `Discover` with the intention that the new types would appear before we fully switch over, then delete this after.

I was starting with this change: https://github.com/getsentry/sentry/pull/37133/files but it got crazy! Our new `Replay` type isn't compatible with `Event`, and i'd like to rename `event` everywhere too. So I was working on a compatibility PR -> https://github.com/getsentry/sentry/pull/37189
Basically the idea is to introduce the new types, and convert as much of the `EventTransaction` fields into the new thing. We've got `ReplayReader` to leverage for the job.

**Review by billyvg — APPROVED** [review]
